adls-plugins/src/main/java/co/cask/hydrator/plugin/sink/ADLSBatchSink.java

```diff
@@ -128,18 +151,60 @@ public void transform(StructuredRecord input, Emitter<KeyValue<AvroKey<GenericRe
     @Macro
     public String fileSystemProperties;
 
-    @Description("Output schema for the output files.")
+    @Description("The format of output files.")
+    @Macro
+    private String outputFormat;
+
+    @Nullable
+    @Description("Output schema of the JSON document. Required for avro output format. " +
+      "If left empty for text output format, the schema of input records will be used." +
+      "This must be a subset of the schema of input records. " +
+      "Fields of schema types ARRAY, MAP, RECORD and UNION in input records cannot be present " +
```

Comment:
Fields of type ARRAY, MAP, and RECORD are not supported with the text format. Fields of type UNION are only supported if they represent a nullable type.